web/api/v1/api.go

```diff
@@ -1351,6 +1352,25 @@ func (api *API) serveTSDBStatus(*http.Request) apiFuncResult {
 	}, nil, nil, nil}
 }
 
+type walReplayStatus struct {
+	First   int  `json:"first"`
+	Last    int  `json:"last"`
+	Read    int  `json:"read"`
+	Started bool `json:"started"`
+	Done    bool `json:"done"`
+}
+
+func (api *API) serveWALReplayStatus(w http.ResponseWriter, r *http.Request) {
+	httputil.SetCORS(w, api.CORSOrigin, r)
+	api.respond(w, walReplayStatus{
+		First:   tsdb.WALReplayStatus.First,
+		Last:    tsdb.WALReplayStatus.Last,
```

Comment:
As mentioned before, I would suggest actually passing this through API struct and add mutex (sync.Lock) to avoid undefined behaviours and even crashes.